### PR TITLE
Math Magicians:  Calculator Button Events

### PR DESCRIPTION
**Pull request by isaacpitwa**
## Math Magicians:  Calculator Button Events
- Copy the files calculate.js and operate.js into a logic/ directory in your project
- Analyze the files calculate.js and operate.js, be sure you understand what they do (note: you need to add [big.js](https://www.npmjs.com/package/big-js) as a dependency)
- Import the files in your Calculator component.
- Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component.
- Tested Application

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-unused-state */
```
Comment (micheaol) [inline src/components/calculator.js]: Please ensure that linters are not disabled.  Linters are tools that analyze your source code to flag programming errors, bugs, stylistic errors, and suspicious constructs(source: [Wikipedia](https://en.wikipedia.org/wiki/Lint_(software))). Kindly enable the disabled linters.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
```
Comment (micheaol) [inline src/logic/calculate.js:17]: Your app is not functioning as expected. The `a/c` button is not responding to click. Kindly fix that.

image

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @isaacpitwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mikethreels — APPROVED** [review]
Hi Isaac, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**